### PR TITLE
Pass public ssh key to ignition file

### PR DESCRIPTION
**Pull request by oshercc**

### CODE REVIEW

File: internal/bminventory/inventory.go [hunk 1]
```diff
@@ -162,7 +163,7 @@ func (b *bareMetalInventory) createImageJob(ctx context.Context, id string) erro
 								},
 								{
 									Name:  "IGNITION_CONFIG",
-									Value: fmt.Sprintf(ignitionConfigFormat, b.InventoryURL, b.InventoryPort),
+									Value: fmt.Sprintf(ignitionConfigFormat, b.GetUserSshKey(id), b.InventoryURL, b.InventoryPort),
```
Comment (filanov) [inline internal/bminventory/inventory.go]: please make it a function.

File: internal/bminventory/inventory.go [hunk 2]
```diff
@@ -185,6 +186,21 @@ func (b *bareMetalInventory) createImageJob(ctx context.Context, id string) erro
 	return nil
 }
 
+func (b *bareMetalInventory) GetUserSshKey(id string)  string{
```
Comment (filanov) [inline internal/bminventory/inventory.go]: i think that the parameter name should be `clusterId` it's not clear what is the id from the name.

File: deploy/bm-inventory.yaml [hunk 3]
```diff
@@ -14,7 +14,7 @@ spec:
     spec:
       containers:
         - name: bm-inventory
-          image: REPLACE_IMAGE
+          image: quay.io/oscohen/bm-inventory:latest
```
[the remark this answers is not in the document]
Comment (filanov) [reply deploy/bm-inventory.yaml]: there i a script that replace it. don't do it manually

File: internal/bminventory/inventory.go [hunk 4]
```diff
@@ -185,6 +186,25 @@ func (b *bareMetalInventory) createImageJob(ctx context.Context, id string) erro
 	return nil
 }
 
+func (b *bareMetalInventory) FormatIgnitionFile(ClusterId string)  string{
+	return fmt.Sprintf(ignitionConfigFormat, b.GetUserSshKey(ClusterId), b.InventoryURL, b.InventoryPort)
+}
+
+func (b *bareMetalInventory) GetUserSshKey(ClusterId string)  string{
+	var cluster models.Cluster
+	b.db.First(&cluster, "id = ?", ClusterId)
```
Comment (filanov) [inline internal/bminventory/inventory.go]: where do you check for errors?

File: internal/bminventory/inventory.go [hunk 5]
```diff
@@ -185,6 +187,28 @@ func (b *bareMetalInventory) createImageJob(ctx context.Context, id string) erro
...
 }
 
+func (b *bareMetalInventory) FormatIgnitionFile(ClusterId string)  string{
+	return fmt.Sprintf(ignitionConfigFormat, b.GetUserSshKey(ClusterId), b.InventoryURL, b.InventoryPort)
+}
+
+func (b *bareMetalInventory) GetUserSshKey(clusterId string)  string{
+	var cluster models.Cluster
+	if err := b.db.First(&cluster, "id = ?", clusterId); err != nil {
+		errors.Wrap(err, "cluster not found")
```
Comment (filanov) [inline internal/bminventory/inventory.go]: you are not doing anything with the error
the ignition is wrong but you continue anyway?

File: internal/bminventory/inventory.go [hunk 6]
```diff
@@ -185,6 +187,28 @@ func (b *bareMetalInventory) createImageJob(ctx context.Context, id string) erro
 	return nil
 }
 
+func (b *bareMetalInventory) FormatIgnitionFile(ClusterId string) string {
```
Comment (filanov) [inline internal/bminventory/inventory.go]: change the function to return string and error
then use this function above the creation of the job and check if it failed. if it is then return an error

File: internal/bminventory/inventory.go [hunk 7]
```diff
@@ -185,6 +187,28 @@ func (b *bareMetalInventory) createImageJob(ctx context.Context, id string) erro
...
+	var cluster models.Cluster
+	if err := b.db.First(&cluster, "id = ?", clusterId); err != nil {
+		return errors.Wrap(err, "cluster not found")
+	}
+
+	if cluster.SSHPublicKey == "" {
+		return ""
+	}
+
+	User := fmt.Sprintf(`,{
```
Comment (filanov) [inline internal/bminventory/inventory.go]:
```suggestion
	return fmt.Sprintf(`,{
```

**Comment by filanov** [comment]
i think it will be a better option if you get the cluster in the beginning of DownloadClusterISO it's missing the validation there anyway.
then pass the cluster to createImageJob then you don't have to validate any get there. and you will use this cluster struct anyway for other parameters in the future

File: internal/bminventory/inventory.go [hunk 8]
```diff
@@ -185,6 +187,21 @@ func (b *bareMetalInventory) createImageJob(ctx context.Context, id string) erro
 	return nil
 }
 
+func (b *bareMetalInventory) FormatIgnitionFile(cluster models.Cluster) string {
```
Comment (filanov) [inline internal/bminventory/inventory.go]: user a pointer.

Comment (filanov) [inline internal/bminventory/inventory.go]: make it a private function `formatIgnitionFile`

File: internal/bminventory/inventory.go [hunk 9]
```diff
@@ -185,6 +187,21 @@ func (b *bareMetalInventory) createImageJob(ctx context.Context, id string) erro
...
+	return fmt.Sprintf(ignitionConfigFormat, b.GetUserSshKey(cluster), b.InventoryURL, b.InventoryPort)
+}
+
+func (b *bareMetalInventory) GetUserSshKey(cluster models.Cluster) string {
```
Comment (filanov) [inline internal/bminventory/inventory.go]: make it a private function `getUserSshKey`

File: internal/bminventory/inventory.go (hunk 8 above)
Comment (oshercc) [reply internal/bminventory/inventory.go]: user vat is removed

File: internal/bminventory/inventory.go [hunk 10]
```diff
@@ -249,7 +266,12 @@ func (b *bareMetalInventory) DeregisterCluster(ctx context.Context, params inven
 }
 
 func (b *bareMetalInventory) DownloadClusterISO(ctx context.Context, params inventory.DownloadClusterISOParams) middleware.Responder {
-	if err := b.createImageJob(ctx, params.ClusterID); err != nil {
+	var cluster *models.Cluster
```
Comment (filanov) [inline internal/bminventory/inventory.go]: here it was find. you have an object and you use its pointer. 
```suggestion
	var cluster models.Cluster
	if err := b.db.First(&cluster, "id = ?", params.ClusterID); err != nil {
```

File: internal/bminventory/inventory.go [hunk 11]
```diff
@@ -249,7 +266,12 @@ func (b *bareMetalInventory) DeregisterCluster(ctx context.Context, params inven
...
+	if err := b.db.First(cluster, "id = ?", params.ClusterID); err != nil {
+		return inventory.NewDownloadClusterISONotFound()
+	}
+
+	if err := b.createImageJob(ctx, cluster); err != nil {
```
Comment (filanov) [inline internal/bminventory/inventory.go]: and then here 
```suggestion
	if err := b.createImageJob(ctx, &cluster); err != nil {
```

**Pull request merged**